pxtcompiler/emitter/decompiler.ts

```diff
@@ -1630,12 +1680,23 @@ ${output}</xml>`;
                                 if (arrow.parameters.length < paramDesc.handlerParameters.length) {
                                     for (let i = arrow.parameters.length; i < paramDesc.handlerParameters.length; i++) {
                                         const arg = paramDesc.handlerParameters[i];
-                                        addInput(getDraggableVariableBlock("HANDLER_DRAG_PARAM_" + arg.name, arg.name));
+                                        if (attributes.draggableParameters === "reporter") {
+                                            addInput(mkDraggableReporterValue("HANDLER_DRAG_PARAM_" + arg.name, arg.name, arg.type));
+                                        } else {
+                                            addInput(getDraggableVariableBlock("HANDLER_DRAG_PARAM_" + arg.name, arg.name));
+                                        }
                                     }
                                 }
+                                if (attributes.draggableParameters === "reporter") {
+                                    env.localReporters.push(paramDesc.handlerParameters);
```

Comment:
Could you add a comment explaining that this is for the `getStatementBlock()` below?